### DISCUSSION TITLE
Next release

### DISCUSSION THREAD

**Discussion by Serrof** [Hipparchus-Math/hipparchus#342] [General]
Hi all,

Orekit is leanining towards a major release after the upcoming 12.1 and I think Hipparchus should do the same. We already have two issues in the 4.0 milestone. Any objections?

Cheers,
Romain.

**Comment by maisonobe** [comment]
Yes, I think we should release something.
I am however working (from time to time) on issue [1388](https://gitlab.orekit.org/orekit/orekit/-/issues/1388) in Orekit which is in fact due to Hipparchus.
Another point, but it is really a long shot, would be to revamp the optimizers API.
I really hate this `OptimizationData` marker interface and would like to drop it and use something more in line with what Evan did with `LeastSquaresBuilder`.

**Comment by Serrof** [comment]
I'm upping this thread. I believe that pending @mgrutten 's work on #363, we're ready for Hipparchus 4.0?

**Reply by maisonobe** [reply]
You are right.
I did not make significant progress on [1388](https://gitlab.orekit.org/orekit/orekit/-/issues/1388), and could not start looking at optimizers, so this will be for a later release.

**Comment by Serrof** [comment]
@maisonobe I just merged Mark's work, so I think we're all set for release
